src/System.IO.Compression.Brotli/src/System/IO/Compression/dec/BrotliStream.Decompress.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression
+{
+    public partial class BrotliStream : Stream
+    {
+        private BrotliDecoder _decoder;
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            ValidateParameters(buffer, offset, count);
+            return ReadCore(new Span<byte>(buffer, offset, count));
+        }
+
+        public override int Read(Span<byte> destination)
+        {
+            if (GetType() != typeof(BrotliStream))
+            {
+                // BrotliStream is not sealed, and a derived type may override Read(byte[], int, int) without also
+                // overriding Read(Span<byte>). In that case, this Read(Span<byte>) overload
+                // should use the behavior of Read(byte[],int,int) overload.
```

Comment:
A few thoughts:
1) BrotliStream could be made sealed.  What's the downside?  Are there real scenarios we care about where someone might derive from it and override members?  If we make it sealed, we can avoid this type check, and the JIT is more likely to be able to devirtualize.
2) BrotliStream is being introduced at the same time that the new span/memory-based virtuals are being introduced.  I don't think we need this type check.  We've had this in types that existed prior to the new methods, because if `Read(byte[], ...)` was overridden, we'd want to make sure that we used that logic from the new `Read(Span)` method, but for someone to derive from BrotliStream, they would explicitly be choosing not to also override `Read(Span)`, at which point that's on them.  My only caveat to this would be if we're shipping a downlevel BrotliStream, such that someone might derive from it but be unable to override the new members.

Reply:
These are almost certainly unnecessary, regardless of the decision of sealed vs non-sealed.  They're in here to accommodate the overriding case but as you said in your point 2, that's on them to not also override the Span overloads.

So either way we can remove this code. As to whether we make BrotliStream sealed... I don't have a strong opinion. I can think of no situation where anyone would want to inherit from BrotliStream just like I can think of no reason why anyone would want to inherit from DeflateStream. So if the guideline is making things sealed when we can, then it should probably be sealed.